Quantum-Control-Applications/ESR/configuration.py

```diff
@@ -0,0 +1,537 @@
+
+# Pi_half pulse parameters
+pi_half_len = int(pi_len/2)  # in units of ns
+pi_half_amp = pi_amp  # in units of volts
+pi_half_wf = (pi_half_amp * (gaussian(pi_half_len, pi_half_len / 5) -
+                             gaussian(pi_half_len, pi_half_len / 5)[-1])).tolist()  # waveform
+minus_pi_half_wf = ((-1) * pi_half_amp * (gaussian(pi_half_len, pi_half_len / 5) -
+                             gaussian(pi_half_len, pi_half_len / 5)[-1])).tolist()  # waveform
+
+# Subtracted Gaussian pulse parameters
+gauss_amp = 0.3  # The gaussian is used when calibrating pi and pi_half pulses
+gauss_len = 20  # The gaussian is used when calibrating pi and pi_half pulses
+gauss_wf = (gauss_amp * (gaussian(gauss_len, gauss_len / 5) -
+                         gaussian(gauss_len, gauss_len / 5)[-1])).tolist()  # waveform
+
+# Note: a subtracted Gaussian pulse has a more narrow spectral density than a regular gaussian
+# it becomes useful in short pulses to reduce leakage to higher energy states
+
+# Flux:
+square_flux_amp = 0.3
+minus_square_flux_amp = -0.3
+triangle_flux_amp = 0.3
+triangle_wf = [triangle_flux_amp * i/7 for i in range(8)] + [triangle_flux_amp * (1 - i/7) for i in range(8)]
+
+initialization_len = 1000  # in ns
+activation_len = 500  # in ns
+
+config = {
+
+    'version': 1,
+
+    'controllers': {
+
+        'con1': {
+
+            'type': 'opx1',
+
+            'analog_outputs': {
+                1: {'offset': +0.0},  # qubit I
+                2: {'offset': +0.0},  # qubit Q
+                3: {'offset': +0.0},  # Resonator I
+                4: {'offset': +0.0},  # Resonator Q
+                5: {'offset': +0.0},
+                6: {'offset': +0.0},
+            },
+
+            'digital_outputs': {
+                1: {},
+                2: {},
+                3: {},
+                4: {},
+                5: {},
+                10: {},
+            },
+
+            'analog_inputs': {
+                1: {'offset': 0.0, 'gain_db': 0},  # I from down conversion
+                2: {'offset': 0.0, 'gain_db': 0},  # Q from down conversion
+            },
+        },
+    },
+
+    'elements': {
+
+        'ensemble': {
+            'mixInputs': {
+                'I': ('con1', 1),
+                'Q': ('con1', 2),
+                'lo_frequency': ensemble_LO,
+                'mixer': 'mixer_ensemble',
+            },
+            'intermediate_frequency': ensemble_if,
+            'operations': {
+                'const': 'const_pulse',
+                'saturation': 'saturation_pulse',
+                'gaussian': 'gaussian_pulse',
+                'pi': 'pi_pulse',
+                'pi_half': 'pi_half_pulse',
+                'x180': 'x180_pulse',
+                'x90': 'x90_pulse',
+                '-x90': '-x90_pulse',
+                'y180': 'y180_pulse',
+                'y90': 'y90_pulse',
+                '-y90': '-y90_pulse',
+            },
+        },
+
+        'ensemble2': {
```

Comment:
what is ensemble2?

Reply:
it was specific to the experiment but now I see that is it not needed and it overcrowds only